### PR TITLE
Update CI configs to v0.4.7

### PR DESCRIPTION
**Pull request by pionbot**
Update lint scripts and CI configs.

### CODE REVIEW

**Comment by scorpionknifes** [comment]
I am taking this one 😃

**Comment by scorpionknifes** [comment]
Not sure if changing this is a good idea. I see tests done for mrand and crand in test pkg.
```
G404: Use of weak random number generator (math/rand instead of crypto/rand) (gosec)
```

**Comment by Sean-Der** [comment]
@scorpionknifes So this is what I have been doing for some of the harder ones. You do w/e you think is best though :)

For `gocognit`, `gosec` and `dupl` I add a `nolint:dupl` (or w/e linter) and cut a ticket

For `godox` I delete the comment and cut a ticket.

For `gochecknoglobals` I make a function that returns the struct.

File: vnet/chunk.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func (f tcpFlag) String() string {
 
 // Generate a base36-encoded unique tag
 // See: https://play.golang.org/p/0ZaAID1q-HN
-var assignChunkTag = func() func() string {
+var assignChunkTag = func() func() string { //nolint:gochecknoglobals
 	var tagCtr uint64
 
 	return func() string {
```
Comment (scorpionknifes) [inline vnet/chunk.go:48]: tagCtr seems to increment globally.

File: vnet/net.go [hunk 2]
```diff
@@ -10,7 +11,26 @@ import (
...
 )
 
-var macAddrCounter uint64 = 0xBEEFED910200
+const (
+	lo0String = "lo0String"
+	udpString = "udp"
+)
+
+var (
+	macAddrCounter                 uint64 = 0xBEEFED910200 //nolint:gochecknoglobals
```
Comment (scorpionknifes) [inline vnet/net.go:20]: macAddrCounter also increments globally

File: vnet/router.go [hunk 3]
```diff
@@ -16,8 +17,19 @@ const (
...
+	errAddressSpaceExhausted         = errors.New("address space exhausted")
+	errNoIPAddrEth0                  = errors.New("no IP address is assigned for eth0")
+)
+
 // Generate a unique router name
-var assignRouterName = func() func() string {
+var assignRouterName = func() func() string { //nolint:gochecknoglobals
 	var routerIDCtr uint64
 
 	return func() string {
```
Comment (scorpionknifes) [inline vnet/router.go:35]: This occurs here as well.

**Review by scorpionknifes — COMMENTED** [review]
A couple of lint for global var that I'm unsure of.

**Review by Sean-Der — APPROVED** [review]

**Comment by Sean-Der** [comment]
amazing work, looked like a frustrating one :)

I can't believe how close we are to being done.

**Pull request merged**